### DISCUSSION TITLE
How to combine Query and Body in one pydantic class

### DISCUSSION THREAD

**Discussion by tomaszZdunek** [fastapi/fastapi#10685] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, Query
from pydantic import Field

VersionedRequestBody = Annotated[Union[RequestBodyVersion1, RequestBodyVersion2], Field(discriminator="api_version")]

class RequestBodyVersion1(BaseModel):
    api_version: Literal["1"] = Field(default=Query(..., alias="api-version"))
    some_field: int = 1


class RequestBodyVersion1(BaseModel):
    api_version: Literal["2"] = Field(default=Query(..., alias="api-version"))
    some_field: int = 2

@router_openai.post("/test")
async def versioned_input(
    deployment_id: OpenAIDeployment,
    body: VersionedRequestBody,
    api_version: OpenAIApiVersion = Query(..., alias="api-version")
) -> Any:
    return body
```

### Description

I want to have versioned body model, but the issue is that my version is in query, not as a part of body. I cannot change this, this format of query and body is required by external process. I tried multiple approach, but nothing works out of the box and all solutions I found requires dirty manual changes in documentation and a lot of code duplications.

### Operating System

macOS

### FastAPI Version

0.78.0

### Pydantic Version

1.10.13

### Python Version

3.9.13

**Comment by YuriiMotov** [comment]
Let's close this for now to keep the discussions clean and organized.
If you’ve found a solution, please share it here.
If you still need help, feel free to let us know.
